src/PrimeTower.cpp

```diff
@@ -224,4 +242,20 @@ void PrimeTower::subtractFromSupport(SliceDataStorage& storage)
 }
 
 
+void PrimeTower::gotoStartLocation(const SliceDataStorage& storage, LayerPlan& gcode_layer, const int extruder_nr) const
+{
+    int current_start_location_idx = ((extruder_nr + 1) * gcode_layer.getLayerNr()) % number_of_prime_tower_start_locations;
```

Comment:
Since `gcode_layer.getLayerNr()` can result in a negative layer index, this integer can become negative. That then results in a segfault in the line below when looking it up in the `prime_tower_start_locations` vector.

To fix, I suggest to do a positive-only modulus wraparound: `((x % n) + n) % n`

Reply:
Improved this for negative layers in d4957ca1